src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/generator/BookingsReportGenerator.kt

```diff
@@ -34,22 +34,22 @@ class BookingsReportGenerator : ReportGenerator<BookingsReportDataAndPersonInfo,
         registeredSexOffender = booking.registeredSexOffender.toYesNo(),
         historyOfSexualOffence = booking.historyOfSexualOffence.toYesNo(),
         concerningSexualBehaviour = booking.concerningSexualBehaviour.toYesNo(),
-        needForAccessibleProperty = booking.needForAccessibleProperty,
+        needForAccessibleProperty = booking.needForAccessibleProperty.toYesNo(),
         historyOfArsonOffence = booking.historyOfArsonOffence.toYesNo(),
         concerningArsonBehaviour = booking.concerningArsonBehaviour.toYesNo(),
-        dutyToReferMade = booking.dutyToReferMade,
+        dutyToReferMade = booking.dutyToReferMade.toYesNo(),
         dateDutyToReferMade = booking.dateDutyToReferMade,
         dutyToReferLocalAuthorityAreaName = booking.dutyToReferLocalAuthorityAreaName,
-        isReferralEligibleForCas3 = booking.referralEligibleForCas3,
+        isReferralEligibleForCas3 = booking.referralEligibleForCas3.toYesNo(),
         referralEligibilityReason = booking.referralEligibilityReason,
         probationRegion = booking.probationRegionName,
         pdu = booking.pdu,
         localAuthority = booking.localAuthorityAreaName,
         town = booking.town,
         postCode = booking.postCode,
         crn = booking.crn,
-        offerAccepted = booking.confirmationId != null,
-        isCancelled = booking.cancellationId != null,
+        offerAccepted = (booking.confirmationId != null).toYesNo().toString(),
```

Comment:
i guess the toString() is to ensure a non null value is never returned. Maybe toYesNo()!! would make more sense here given we know it can never be null? (this pattern is in a few places). Or we could add a version of toYesNo that operates on non-null booleans?

Reply:
Yes I changed it to toYesNo()!!